public/app/features/logs/components/logParser.ts

```diff
@@ -103,13 +120,12 @@ function shouldRemoveField(field: Field, index: number, row: LogRowModel) {
   ) {
     return true;
   }
-  // hidden field
-  if (field.config.custom?.hidden) {
-    return true;
-  }
-  // field that has empty value (we want to keep 0 or empty string)
-  if (field.values[row.rowIndex] == null) {
+
+  // first string-field is the log-line
+  const firstStringFieldIndex = row.dataFrame.fields.findIndex((f) => f.type === FieldType.string);
+  if (firstStringFieldIndex === index) {
```

Comment:
this is the "real" new code in the PR, the check for the log-line-field (first string-field in the dataframe)